enumerables.rb

```diff
@@ -0,0 +1,109 @@
+# Implementing  ruby enum methods
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    length.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    length.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    temp = []
+    to_a.my_each { |item| temp.push(item) if yield(item) }
+    temp
+  end
+
+  def my_all?(arg = nil)
```

Comment:
- [ ]  **#my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex. The following code should return True. Instead, it is returning false.** 

```
expect(words.my_all?(/d/))
```